documentation/RELEASE_NOTES.md

```diff
@@ -18,6 +18,33 @@ should also be read to understand what has changed since earlier releases.
 <!-- Insert new items immediately below here ... -->
 
 
+### Timeouts for Unit Test Programs
+
+Most unit test programs that are run by the EPICS build system using
+`make runtests` or `make tapfiles` are now only allowed to run for a limited
+time period. If a test program runs for more than the timeout period (by
+default 5 minutes, 300 seconds) without exiting it will be killed after that
+period has elapsed. The limit can be extended (or shortened) for individual
+tests as necessary by adding a line to the Makefile which runs them, setting
+the environment variable `EPICS_UNITTEST_TIMEOUT` to a value in seconds while
+building the test (`.t`) script, like this:
+
+```
+  TESTSCRIPTS_HOST += hourLongTest.t
+  hourLongTest.t: export EPICS_UNITTEST_TIMEOUT=3600
+```
+
+The time limit only applies to tests that use `makeTestfile.pl` to generate a
+test (`.t`) script that runs the actual test program. Tests which are provided
+as a `.plt` script should implement their own timeout if they can hang while
+running in a Continuous Integration system (the Base "netget" test now does
+this in a way that works on Windows as well as Posix hosts).
+
+The same environment variable can be set while running the tests to override
+the time limit given in the Makefile, but it cannot be configured to provide a
+separate limit for each program when using `make runtests`. This feature may
+be useful for debugging purposes.
```

Comment:
Is this statement about overriding correct?  It looks to me it would not be possible for CI environment to override a hard coded timeout.

```make
all: target

target: export FOO=baz
target:
        echo "FOO|$$FOO|${FOO}"
        env| grep FOO
```

```sh
$ FOO=bar make
echo "FOO|$FOO|baz"
FOO|baz|baz
env| grep FOO
FOO=baz
```

Reply:
Your Makefile is too simple because the `target` target isn't creating a script that later gets run from a different make rule. EPICS has one set of rules for generating the `.t` scripts and another for running them and creating the `.tap` files. My notes recommend using the target-specific syntax so the variable will only be set when building the `.t` scripts, not when running them:
```makefile
all: build run

build: export FOO=baz

build:
	echo "build: FOO|$$FOO|${FOO}"
run:
	echo "run: FOO|$$FOO|${FOO}"
```

```
$ FOO=bar make
echo "build: FOO|$FOO|baz"
build: FOO|baz|baz
echo "run: FOO|$FOO|bar"
run: FOO|bar|bar
```

Reply:
> Your Makefile is too simple ...

Ah, if only this could be true more often...

Still, I accept your point.